# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1532089547 14400
#      Fri Jul 20 08:25:47 2018 -0400
# Node ID 178f6d18c9a86c1bd8e00256f9f0394f30779853
# Parent  06a3eaf60b237c7cd7588138b831b08c2576988e
maint: reorganize rules for creating images.mk


* doc/interpreter/module.mk: Include images.mk here instead of in
Makefile.am.

* doc/interpreter/module.mk: Move rule for generating images.mk here
from doc/refcard/module.mk.  Use %reldir% in more places.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -206,17 +206,16 @@ libgnu/libgnu.la: oct-conf-post.h
 	cd libgnu && $(MAKE) $(AM_MAKEFLAGS) all
 
 include liboctave/module.mk
 include libinterp/module.mk
 include libgui/module.mk
 include src/module.mk
 include scripts/module.mk
 include doc/module.mk
-include doc/interpreter/images.mk
 include etc/module.mk
 include examples/module.mk
 include m4/module.mk
 include test/module.mk
 include build-aux/module.mk
 
 DIST_HOOKS := \
   doc-interpreter-dist-hook \
diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -320,16 +320,17 @@ DOC_TARGETS += \
   %reldir%/octave.pdf \
   $(OCTAVE_HTML_STAMP) \
   $(HTMLDIR_IMAGES) \
   $(HTMLDIR_CSS)
 
 if AMCOND_BUILD_QT_DOCS
 DOC_TARGETS += \
   $(OCTAVE_QTHELP_FILES)
+
 endif
 
 ## Distribute both OCTAVE_CSS and HTMLDIR_CSS so that the rules for
 ## building HTMLDIR_CSS work properly.
 
 doc_EXTRA_DIST += \
   $(BUILT_OCTAVE_TEXI_SRC) \
   $(srcdir)/%reldir%/octave.info \
@@ -419,16 +420,23 @@ doc_MAINTAINERCLEANFILES += \
 
 ## The TeX software suite is used to create both PDF and PS output formats.
 ## In order to avoid race conditions between simultaneous TeX commands, the
 ## PDF and PS builds are forced to run serially through the following rule.
 %reldir%/octave.pdf: %reldir%/octave.ps
 
 endif
 
+## These actions should happen even if we are not building docs
+
+include doc/interpreter/images.mk
+
+$(srcdir)/%reldir%/images.mk: $(srcdir)/%reldir%/config-images.sh $(srcdir)/%reldir%/images.awk $(srcdir)/%reldir%/images
+	$(AM_V_GEN)$(SHELL) $(srcdir)/%reldir%/config-images.sh $(top_srcdir)
+
 DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
 
 ## The doc-cache file can be built without TeX but it does require
 ## makeinfo, but that is needed to display function docstrings at the
 ## Octave command line.  The macros.texi file must also be installed
 ## to display docstrings at the command line.
 
 doc_MAINTAINERCLEANFILES += \
@@ -440,29 +448,29 @@ octetc_DATA += \
 
 %reldir%/doc-cache: $(DOCSTRING_FILES) %reldir%/mk-doc-cache.pl | $(OCTAVE_INTERPRETER_TARGETS) %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(PERL) $(srcdir)/%reldir%/mk-doc-cache.pl $(srcdir) $(srcdir)/%reldir%/macros.texi $(DOCSTRING_FILES) > $@-t && \
 	mv $@-t $@
 
 %reldir%/undocumented_list:
 	rm -f $@-t $@
-	-cd $(srcdir)/doc/interpreter; $(PERL) ./doccheck/mk_undocumented_list > $(@F)-t
+	-cd $(srcdir)/%reldir%; $(PERL) ./doccheck/mk_undocumented_list > $(@F)-t
 	mv $@-t $@
 .PHONY: %reldir%/undocumented_list
 
 SPELLCHECK_FILES = $(MUNGED_TEXI_SRC:.texi=.scheck)
 
 %.scheck: %.texi | %reldir%/$(octave_dirstamp)
-	cd $(srcdir)/doc/interpreter; ./doccheck/spellcheck $(<F) > $(@F)-t
+	cd $(srcdir)/%reldir%; ./doccheck/spellcheck $(<F) > $(@F)-t
 	mv $@-t $@
 	[ -s $@ ] || rm -f $@
 
 spellcheck: $(SPELLCHECK_FILES)
-	@cd $(srcdir)/doc/interpreter ; \
+	@cd $(srcdir)/%reldir% ; \
 	if ls *.scheck >/dev/null 2>&1 ; then \
 		echo "Spellcheck failed"; \
 		echo "Review the following files:"; \
 		ls *.scheck ; \
 		exit 1 ; \
 	else \
 		echo "Spellcheck passed"; \
 	fi
diff --git a/doc/refcard/module.mk b/doc/refcard/module.mk
--- a/doc/refcard/module.mk
+++ b/doc/refcard/module.mk
@@ -39,19 +39,16 @@ DOC_TARGETS += \
 %reldir%/refcard-legal.ps: %reldir%/refcard-legal.dvi
 	-$(AM_V_DVIPS)$(DVIPS) $(AM_V_texinfo) -T 14in,8.5in -o $@ $<
 
 %reldir%/refcard-letter.pdf: %reldir%/refcard.tex
 %reldir%/refcard-letter.dvi: %reldir%/refcard.tex
 %reldir%/refcard-letter.ps: %reldir%/refcard-letter.dvi
 	-$(AM_V_DVIPS)$(DVIPS) $(AM_V_texinfo) -T 11in,8.5in -o $@ $<
 
-$(srcdir)/doc/interpreter/images.mk: $(srcdir)/doc/interpreter/config-images.sh $(srcdir)/doc/interpreter/images.awk $(srcdir)/doc/interpreter/images
-	$(AM_V_GEN)$(SHELL) $(srcdir)/doc/interpreter/config-images.sh $(top_srcdir)
-
 $(refcard_DVI) : %.dvi : %.tex | %reldir%/$(octave_dirstamp)
 	-$(AM_V_TEX)cd $(@D) && \
 	TEXINPUTS="$(abs_top_srcdir)/doc/refcard:$(TEXINPUTS):" \
 	$(TEX) $(<F) $(AM_V_texidevnull)
 
 $(refcard_PDF) : %.pdf : %.tex | %reldir%/$(octave_dirstamp)
 	-$(AM_V_PDFTEX)cd $(@D) && \
 	TEXINPUTS="$(abs_top_srcdir)/doc/refcard:$(TEXINPUTS):" \
